### DISCUSSION TITLE
platform->ModuleNotFoundError: No module named 'langchain.globals'.frontend ->Database is unavailable - Sleeping...

### DISCUSSION THREAD

**Discussion by jianyue12** [reworkd/AgentGPT#1558] [Q&A]
The error in the title occurred during installation. How to fix it in vscode&"D:\Program Files\AgentGPT"?
docker->![image](https://github.com/reworkd/AgentGPT/assets/109535184/d362bf76-4323-4ba1-9372-5ce4dd57bc53)
The following is the VScode terminal content:
PS D:\Program Files\AgentGPT> ./setup.bat

up to date in 2s

37 packages are looking for funding
  run `npm fund` for details

> agentgpt-cli@1.0.0 start
> node src/index.js

 █████╗  ██████╗ ███████╗███╗   ██╗████████╗ ██████╗ ██████╗ ████████╗
██╔══██╗██╔════╝ ██╔════╝████╗  ██║╚══██╔══╝██╔════╝ ██╔══██╗╚══██╔══╝
███████║██║  ███╗█████╗  ██╔██╗ ██║   ██║   ██║  ███╗██████╔╝   ██║   
██╔══██║██║   ██║██╔══╝  ██║╚██╗██║   ██║   ██║   ██║██╔═══╝    ██║   
██║  ██║╚██████╔╝███████╗██║ ╚████║   ██║   ╚██████╔╝██║        ██║   
╚═╝  ╚═╝ ╚═════╝ ╚══════╝╚═╝  ╚═══╝   ╚═╝    ╚═════╝ ╚═╝        ╚═╝   

Welcome to the AgentGPT CLI! This CLI will generate the required .env files.
Copies of the generated envs will be created in `./next/.env` and `./platform/.env`.

Existing ./next/env file found. Validating...
? How will you be running AgentGPT? 🐋 Docker-compose (Recommended)
time="2024-05-25T11:25:37+08:00" level=warning msg="D:\\Program Files\\AgentGPT\\docker-compose.yml: `version` is obsolete"
[+] Building 2.1s (22/22)                                                                                                                   docker:default
[+] Building 5.1s (37/37) FINISHED                                                                                                          docker:default
 => [agentgpt_db internal] load build definition from Dockerfile                                                                                      0.1s
 => => transferring dockerfile: 98B                                                                                                                   0.0s 
 => [frontend internal] load build definition from Dockerfile                                                                                         0.1s 
 => => transferring dockerfile: 919B                                                                                                                  0.0s 
 => [frontend internal] load metadata for docker.io/library/node:19-alpine                                                                            0.0s 
 => [agentgpt_db internal] load metadata for docker.io/library/mysql:8.0                                                                              0.0s 
 => [agentgpt_db internal] load .dockerignore                                                                                                         0.0s 
 => => transferring context: 2B                                                                                                                       0.0s 
 => [frontend internal] load .dockerignore                                                                                                            0.1s 
 => => transferring context: 111B                                                                                                                     0.0s
 => [agentgpt_db 1/2] FROM docker.io/library/mysql:8.0                                                                                                0.1s 
 => [agentgpt_db internal] load build context                                                                                                         0.1s 
 => => transferring context: 31B                                                                                                                      0.0s 
 => [frontend  1/10] FROM docker.io/library/node:19-alpine                                                                                            0.0s 
 => [frontend internal] load build context                                                                                                            0.1s
 => => transferring context: 23.51kB                                                                                                                  0.1s 
 => [agentgpt_db 2/2] ADD setup.sql /docker-entrypoint-initdb.d                                                                                       0.1s
 => [agentgpt_db] exporting to image                                                                                                                  0.2s 
 => => exporting layers                                                                                                                               0.1s 
 => => writing image sha256:a107521ed833cbdb0730d5305ca13d2dbacf6bab49dd40bba2d5cdb8b9083b33                                                          0.0s 
 => => naming to docker.io/library/mysql:8.0                                                                                                          0.0s 
 => CACHED [frontend  2/10] RUN apk add --no-cache netcat-openbsd                                                                                     0.0s 
 => CACHED [frontend  3/10] WORKDIR /next                                                                                                             0.0s 
 => CACHED [frontend  4/10] COPY package*.json ./                                                                                                     0.0s
 => CACHED [frontend  5/10] RUN npm ci                                                                                                                0.0s
 => CACHED [frontend  6/10] COPY wait-for-db.sh /usr/local/bin/wait-for-db.sh                                                                         0.0s 
 => CACHED [frontend  7/10] RUN chmod +x /usr/local/bin/wait-for-db.sh                                                                                0.0s 
 => CACHED [frontend  8/10] COPY . .                                                                                                                  0.0s 
 => CACHED [frontend  9/10] COPY entrypoint.sh /                                                                                                      0.0s 
 => CACHED [frontend 10/10] RUN apk add --no-cache dos2unix netcat-openbsd     && dos2unix /entrypoint.sh                                             0.0s 
 => [frontend] exporting to image                                                                                                                     0.1s 
 => => exporting layers                                                                                                                               0.0s 
 => => writing image sha256:7175bb691e7b4032ecbb25ae1bbdc06fa2a8e401fbac3d31623e1800cc132fea                                                          0.0s 
 => => naming to docker.io/library/agentgpt-frontend                                                                                                  0.0s 
 => [platform internal] load build definition from Dockerfile                                                                                         0.0s 
 => => transferring dockerfile: 796B                                                                                                                  0.0s 
 => [platform internal] load metadata for docker.io/library/python:3.11-slim-buster                                                                   2.6s 
 => [platform internal] load .dockerignore                                                                                                            0.0s 
 => => transferring context: 2.29kB                                                                                                                   0.0s 
 => [platform internal] load build context                                                                                                            0.1s 
 => => transferring context: 11.32kB                                                                                                                  0.0s 
 => [platform prod  1/10] FROM docker.io/library/python:3.11-slim-buster@sha256:c46b0ae5728c2247b99903098ade3176a58e274d9c7d2efeaaab3e0621a53935      0.0s 
 => CACHED [platform prod  2/10] RUN apt-get update && apt-get install -y   default-libmysqlclient-dev   gcc   pkg-config   openjdk-11-jdk   build-e  0.0s 
 => CACHED [platform prod  3/10] RUN pip install poetry==1.4.2                                                                                        0.0s 
 => CACHED [platform prod  4/10] RUN poetry config virtualenvs.create false                                                                           0.0s 
 => CACHED [platform prod  5/10] COPY pyproject.toml /app/src/                                                                                        0.0s 
 => CACHED [platform prod  6/10] WORKDIR /app/src                                                                                                     0.0s 
 => CACHED [platform prod  7/10] RUN poetry install --only main                                                                                       0.0s 
 => CACHED [platform prod  8/10] RUN apt-get purge -y   g++   gcc   pkg-config   && rm -rf /var/lib/apt/lists/*                                       0.0s 
 => CACHED [platform prod  9/10] COPY . /app/src/                                                                                                     0.0s 
 => CACHED [platform prod 10/10] RUN poetry install --only main                                                                                       0.0s 
 => [platform] exporting to image                                                                                                                     0.0s 
 => => exporting layers                                                                                                                               0.0s 
 => => writing image sha256:ec608cff79b59fa706583a5fce332635de0fc67e97bb7a56548038ef2ca76fb2                                                          0.0s 
 => => naming to docker.io/library/agentgpt-platform                                                                                                  0.0s 
[+] Running 0/3
 - Container agentgpt_db  Created                                                                                                                    20.6s 
 - Container frontend     Created                                                                                                                    20.6s 
 - Container platform     Created                                                                                                                    20.5s 
Attaching to agentgpt_db, frontend, platform
agentgpt_db  | 2024-05-25 03:26:05+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.27-1debian10 started.
frontend     | dos2unix: converting file wait-for-db.sh to Unix format...
frontend     | dos2unix: converting file .env.temp to Unix format...
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25 03:26:05+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
agentgpt_db  | 2024-05-25 03:26:05+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.27-1debian10 started.
platform     | INFO:     Will watch for changes in these directories: ['/app/src']
platform     | INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
platform     | INFO:     Started reloader process [1] using WatchFiles
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:07.473039Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.27) starting as process 1
agentgpt_db  | 2024-05-25T03:26:07.525450Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
agentgpt_db  | 2024-05-25T03:26:08.935477Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:09.679739Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:09.679813Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1.1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:09.681673Z 0 [Warning] [MY-013595] [Server] Failed to initialize TLS for channel: mysql_main. See below for the description of exact issue.
agentgpt_db  | 2024-05-25T03:26:09.681711Z 0 [Warning] [MY-010069] [Server] Failed to set up SSL because of the following SSL library error: SSL_CTX_set_default_verify_paths failed
agentgpt_db  | 2024-05-25T03:26:09.682824Z 0 [ERROR] [MY-010285] [Server] Failure to parse RSA private key (file exists): /var/lib/mysql//private_key.pem: error:0909006C:PEM routines:get_name:no start line
agentgpt_db  | 2024-05-25T03:26:09.683052Z 0 [ERROR] [MY-010119] [Server] Aborting
agentgpt_db  | 2024-05-25T03:26:10.940909Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.27)  MySQL Community Server - GPL.
agentgpt_db exited with code 0
agentgpt_db  | 2024-05-25 03:26:12+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
agentgpt_db  | 2024-05-25 03:26:12+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.27-1debian10 started.
agentgpt_db  | 2024-05-25T03:26:13.241290Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.27) starting as process 1
agentgpt_db  | 2024-05-25T03:26:13.252050Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
agentgpt_db  | 2024-05-25T03:26:13.990924Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:14.438554Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:14.438596Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1.1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:14.438901Z 0 [Warning] [MY-013595] [Server] Failed to initialize TLS for channel: mysql_main. See below for the description of exact issue.
agentgpt_db  | 2024-05-25T03:26:14.438926Z 0 [Warning] [MY-010069] [Server] Failed to set up SSL because of the following SSL library error: SSL_CTX_set_default_verify_paths failed
agentgpt_db  | 2024-05-25T03:26:14.438981Z 0 [ERROR] [MY-010285] [Server] Failure to parse RSA private key (file exists): /var/lib/mysql//private_key.pem: error:0909006C:PEM routines:get_name:no start line
agentgpt_db  | 2024-05-25T03:26:14.439166Z 0 [ERROR] [MY-010119] [Server] Aborting
platform     | Process SpawnProcess-1:
platform     | Traceback (most recent call last):
platform     |   File "/usr/local/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
platform     |     self.run()
platform     |   File "/usr/local/lib/python3.11/multiprocessing/process.py", line 108, in run
platform     |     self._target(*self._args, **self._kwargs)
platform     |   File "/usr/local/lib/python3.11/site-packages/uvicorn/_subprocess.py", line 76, in subprocess_started
platform     |     target(sockets=sockets)
platform     |   File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 61, in run
platform     |     return asyncio.run(self.serve(sockets=sockets))
platform     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
platform     |   File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
platform     |     return runner.run(main)
platform     |            ^^^^^^^^^^^^^^^^
platform     |   File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
platform     |     return self._loop.run_until_complete(task)
platform     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
platform     |   File "uvloop/loop.pyx", line 1517, in uvloop.loop.Loop.run_until_complete
platform     |   File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 68, in serve
platform     |     config.load()
platform     |   File "/usr/local/lib/python3.11/site-packages/uvicorn/config.py", line 473, in load
platform     |     self.loaded_app = import_from_string(self.app)
platform     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
platform     |   File "/usr/local/lib/python3.11/site-packages/uvicorn/importer.py", line 24, in import_from_string
platform     |     raise exc from None
platform     |   File "/usr/local/lib/python3.11/site-packages/uvicorn/importer.py", line 21, in import_from_string
platform     |     module = importlib.import_module(module_str)
platform     |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
platform     |   File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
platform     |     return _bootstrap._gcd_import(name[level:], package, level)
platform     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
platform     |   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
platform     |   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
platform     |   File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
platform     |   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
platform     |   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
platform     |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
platform     |   File "/app/src/reworkd_platform/web/application.py", line 11, in <module>
platform     |     from reworkd_platform.web.api.router import api_router
platform     |   File "/app/src/reworkd_platform/web/api/router.py", line 3, in <module>
platform     |     from reworkd_platform.web.api import agent, auth, metadata, models, monitoring
platform     |   File "/app/src/reworkd_platform/web/api/agent/__init__.py", line 2, in <module>
platform     |     from reworkd_platform.web.api.agent.views import router
platform     |   File "/app/src/reworkd_platform/web/api/agent/views.py", line 17, in <module>
platform     |     from reworkd_platform.web.api.agent.agent_service.agent_service_provider import (
platform     |   File "/app/src/reworkd_platform/web/api/agent/agent_service/agent_service_provider.py", line 15, in <module>
platform     |     from reworkd_platform.web.api.agent.agent_service.open_ai_agent_service import (
platform     |   File "/app/src/reworkd_platform/web/api/agent/agent_service/open_ai_agent_service.py", line 4, in <module>
platform     |     from lanarky.responses import StreamingResponse
platform     |   File "/usr/local/lib/python3.11/site-packages/lanarky/__init__.py", line 1, in <module>
platform     |     from .responses import StreamingResponse
platform     |   File "/usr/local/lib/python3.11/site-packages/lanarky/responses/__init__.py", line 1, in <module>
platform     |     from .streaming import StreamingResponse
platform     |   File "/usr/local/lib/python3.11/site-packages/lanarky/responses/streaming.py", line 18, in <module>
platform     |     from lanarky.callbacks import (
platform     |   File "/usr/local/lib/python3.11/site-packages/lanarky/callbacks/__init__.py", line 11, in <module>
platform     |     from .agents import *  # noqa: F401, F403
platform     |     ^^^^^^^^^^^^^^^^^^^^^
platform     |   File "/usr/local/lib/python3.11/site-packages/lanarky/callbacks/agents.py", line 10, in <module>
platform     |     from .base import AsyncLanarkyCallback
platform     |   File "/usr/local/lib/python3.11/site-packages/lanarky/callbacks/base.py", line 7, in <module>
platform     |     from langchain.globals import get_llm_cache
platform     | ModuleNotFoundError: No module named 'langchain.globals'
agentgpt_db  | 2024-05-25T03:26:15.772295Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.27)  MySQL Community Server - GPL.
agentgpt_db exited with code 0
agentgpt_db  | 2024-05-25 03:26:17+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
agentgpt_db  | 2024-05-25 03:26:17+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.27-1debian10 started.
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:19.903974Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.27) starting as process 1
agentgpt_db  | 2024-05-25T03:26:20.042157Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:20.911277Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
agentgpt_db  | 2024-05-25T03:26:21.506720Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:21.506776Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1.1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:21.507452Z 0 [Warning] [MY-013595] [Server] Failed to initialize TLS for channel: mysql_main. See below for the description of exact issue.
agentgpt_db  | 2024-05-25T03:26:21.507499Z 0 [Warning] [MY-010069] [Server] Failed to set up SSL because of the following SSL library error: SSL_CTX_set_default_verify_paths failed
agentgpt_db  | 2024-05-25T03:26:21.507597Z 0 [ERROR] [MY-010285] [Server] Failure to parse RSA private key (file exists): /var/lib/mysql//private_key.pem: error:0909006C:PEM routines:get_name:no start line
agentgpt_db  | 2024-05-25T03:26:21.507867Z 0 [ERROR] [MY-010119] [Server] Aborting
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:23.068118Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.27)  MySQL Community Server - GPL.
agentgpt_db exited with code 0
agentgpt_db  | 2024-05-25 03:26:24+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
agentgpt_db  | 2024-05-25 03:26:24+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.27-1debian10 started.
agentgpt_db  | 2024-05-25T03:26:24.836367Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.27) starting as process 1
agentgpt_db  | 2024-05-25T03:26:24.853432Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:25.128206Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
agentgpt_db  | 2024-05-25T03:26:25.478010Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:25.478089Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1.1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:25.479209Z 0 [Warning] [MY-013595] [Server] Failed to initialize TLS for channel: mysql_main. See below for the description of exact issue.
agentgpt_db  | 2024-05-25T03:26:25.479262Z 0 [Warning] [MY-010069] [Server] Failed to set up SSL because of the following SSL library error: SSL_CTX_set_default_verify_paths failed
agentgpt_db  | 2024-05-25T03:26:25.479361Z 0 [ERROR] [MY-010285] [Server] Failure to parse RSA private key (file exists): /var/lib/mysql//private_key.pem: error:0909006C:PEM routines:get_name:no start line
agentgpt_db  | 2024-05-25T03:26:25.479788Z 0 [ERROR] [MY-010119] [Server] Aborting
agentgpt_db  | 2024-05-25T03:26:26.908531Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.27)  MySQL Community Server - GPL.
frontend     | Database is unavailable - Sleeping...
agentgpt_db exited with code 1
agentgpt_db  | 2024-05-25 03:26:28+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
agentgpt_db  | 2024-05-25 03:26:28+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.27-1debian10 started.
agentgpt_db  | 2024-05-25T03:26:28.603213Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.27) starting as process 1
agentgpt_db  | 2024-05-25T03:26:28.613454Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
agentgpt_db  | 2024-05-25T03:26:28.792841Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:28.999921Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:28.999985Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1.1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:29.000512Z 0 [Warning] [MY-013595] [Server] Failed to initialize TLS for channel: mysql_main. See below for the description of exact issue.
agentgpt_db  | 2024-05-25T03:26:29.000562Z 0 [Warning] [MY-010069] [Server] Failed to set up SSL because of the following SSL library error: SSL_CTX_set_default_verify_paths failed
agentgpt_db  | 2024-05-25T03:26:29.000619Z 0 [ERROR] [MY-010285] [Server] Failure to parse RSA private key (file exists): /var/lib/mysql//private_key.pem: error:0909006C:PEM routines:get_name:no start line
agentgpt_db  | 2024-05-25T03:26:29.000843Z 0 [ERROR] [MY-010119] [Server] Aborting
agentgpt_db  | 2024-05-25T03:26:30.454087Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.27)  MySQL Community Server - GPL.
agentgpt_db  | 2024-05-25 03:26:32+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
agentgpt_db  | 2024-05-25 03:26:32+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.27-1debian10 started.
agentgpt_db  | 2024-05-25T03:26:32.765087Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.27) starting as process 1
agentgpt_db  | 2024-05-25T03:26:32.773706Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
agentgpt_db  | 2024-05-25T03:26:32.945593Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
agentgpt_db  | 2024-05-25T03:26:33.103195Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:33.103225Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1.1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:33.103514Z 0 [Warning] [MY-013595] [Server] Failed to initialize TLS for channel: mysql_main. See below for the description of exact issue.
agentgpt_db  | 2024-05-25T03:26:33.103537Z 0 [Warning] [MY-010069] [Server] Failed to set up SSL because of the following SSL library error: SSL_CTX_set_default_verify_paths failed
agentgpt_db  | 2024-05-25T03:26:33.103595Z 0 [ERROR] [MY-010285] [Server] Failure to parse RSA private key (file exists): /var/lib/mysql//private_key.pem: error:0909006C:PEM routines:get_name:no start line
agentgpt_db  | 2024-05-25T03:26:33.103775Z 0 [ERROR] [MY-010119] [Server] Aborting
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:34.607402Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.27)  MySQL Community Server - GPL.
agentgpt_db  | 2024-05-25 03:26:38+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
agentgpt_db  | 2024-05-25 03:26:38+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.27-1debian10 started.
agentgpt_db  | 2024-05-25T03:26:38.549795Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.27) starting as process 1
agentgpt_db  | 2024-05-25T03:26:38.556897Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
agentgpt_db  | 2024-05-25T03:26:38.814566Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
frontend     | Database is unavailable - Sleeping...
agentgpt_db  | 2024-05-25T03:26:39.416474Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:39.416503Z 0 [Warning] [MY-013746] [Server] A deprecated TLS version TLSv1.1 is enabled for channel mysql_main
agentgpt_db  | 2024-05-25T03:26:39.417232Z 0 [Warning] [MY-013595] [Server] Failed to initialize TLS for channel: mysql_main. See below for the description of exact issue.
agentgpt_db  | 2024-05-25T03:26:39.417258Z 0 [Warning] [MY-010069] [Server] Failed to set up SSL because of the following SSL library error: SSL_CTX_set_default_verify_paths failed
agentgpt_db  | 2024-05-25T03:26:39.417314Z 0 [ERROR] [MY-010285] [Server] Failure to parse RSA private key (file exists): /var/lib/mysql//private_key.pem: error:0909006C:PEM routines:get_name:no start line
agentgpt_db  | 2024-05-25T03:26:39.417510Z 0 [ERROR] [MY-010119] [Server] Aborting
agentgpt_db  | 2024-05-25T03:26:40.673466Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.27)  MySQL Community Server - GPL.
frontend     | Database is unavailable - Sleeping...
Gracefully stopping... (press Ctrl+C again to force)

**Comment by antoniocifu** [comment]
The problem is the lanarky version.
Change to lanarky="0.7.15" in pyproject.toml to prevent higher versions from being installed
